### PR TITLE
Feat #881/color picker in product card

### LINKED ISSUE #881: [FEATURE] Implement SfColorPicker in SfProductCard
**Describe how this feature will help us improve**
This issue is blocked by https://github.com/DivanteLtd/storefront-ui/issues/851
With new design, we need to implement SfColorPicker in SfProductCard

**Additional context**
Mobile: https://www.figma.com/file/N0Ct95cSAoODNv7zYS01ng/Storefront-UI-Design-System?node-id=264%3A566
Desktop: https://www.figma.com/file/N0Ct95cSAoODNv7zYS01ng/Storefront-UI-Design-System?node-id=771%3A306

### PR DESCRIPTION
**Pull request by Baroshem**
# Related issue

Closes #881 

# Scope of work

# Screenshots of visual changes

# Checklist

- [ ] No commented blocks of code left
- [ ] Run tests and docs
- [ ] Self code-reviewed
- [ ] Changes documented 

If applicable:

- [ ] I followed [composition rules](https://docs.storefrontui.io/?path=/story/introduction-contributing-guide-code-guidelines--page) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/vue/src/components/organisms/SfProductCard/SfProductCard.vue [hunk 1]
```diff
@@ -347,6 +386,18 @@ export default {
...
     },
+    handleSelectedColor(colorIndex) {
+      if (this.colors.length > 0) {
+        this.colors.map((color, i) => {
+          if (colorIndex === i) {
+            this.$emit("click:colors", color);
+          }
+        });
+      }
+    },
```
Comment (AdamPawlinski) [inline packages/vue/src/components/organisms/SfProductCard/SfProductCard.vue:397]: I wonder, should the color picker be closed after choosing color on mobile? Maybe adding toggleColorPicker with isMobile condition here, could be good idea? Wdyt?

Comment (AdamPawlinski) [inline packages/vue/src/components/organisms/SfProductCard/SfProductCard.vue:397]: I wonder, should the color picker be closed after choosing color on mobile? Maybe adding toggleColorPicker with isMobile condition here, could be good idea? Wdyt?

**Review by justyna-13 — COMMENTED** [review]
I agree with Adam

**Review by AdamPawlinski — APPROVED** [review]

**Review by justyna-13 — APPROVED** [review]
:+1:

**Pull request merged**